File: src/main/java/cards/Bandaid.java
package cards;

import com.megacrit.cardcrawl.actions.common.GainBlockAction;
import com.megacrit.cardcrawl.actions.common.HealAction;
import com.megacrit.cardcrawl.cards.AbstractCard;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.core.CardCrawlGame;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.localization.CardStrings;
import com.megacrit.cardcrawl.monsters.AbstractMonster;
import enums.AbstractCardEnum;

public class Bandaid extends PainCustomCard {

    public static final String ID = "Bandaid";

    private static final int COST = 1;
    private static final int BLOCK_AMT = 8;
    private static final int HEAL_AMT = 2;
    private static final int BLOCK_UPG_AMT = 2;
    private static final int HEAL_UPG = 1;


    public Bandaid() {
        super(ID, COST, CardType.SKILL, AbstractCardEnum.THE_PAIN_PURPLE, CardRarity.COMMON, CardTarget.SELF);
        this.block = this.baseBlock = BLOCK_AMT;
        this.magicNumber = this.baseMagicNumber = HEAL_AMT;
    }

    @Override
    public void upgrade() {
        if(!this.upgraded) {
            this.upgradeName();
            this.upgradeBlock(BLOCK_UPG_AMT);
            this.upgradeMagicNumber(HEAL_UPG);
        }
    }

    @Override
    public AbstractCard makeCopy() {
        return new Bandaid();
    }

    @Override
    public void use(AbstractPlayer abstractPlayer, AbstractMonster abstractMonster) {
        AbstractDungeon.actionManager.addToBottom(new GainBlockAction(abstractPlayer, abstractPlayer ,this.block));
        AbstractDungeon.actionManager.addToBottom(new HealAction(abstractPlayer, abstractPlayer, this.magicNumber));

    }

}

File: src/main/kotlin/kobting/thepain/cards/Bandaid.kt
package kobting.thepain.cards

import com.megacrit.cardcrawl.actions.common.GainBlockAction
import com.megacrit.cardcrawl.actions.common.HealAction
import com.megacrit.cardcrawl.cards.AbstractCard
import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.dungeons.AbstractDungeon
import com.megacrit.cardcrawl.monsters.AbstractMonster
import kobting.thepain.patches.AbstractCardEnum

class Bandaid :
        PainCustomCard(
                ID,
                COST,
                AbstractCard.CardType.SKILL,
                AbstractCardEnum.THE_PAIN_PURPLE,
                AbstractCard.CardRarity.COMMON,
                AbstractCard.CardTarget.SELF)
{

    companion object {

        @JvmStatic
        val ID = "thepain:Bandaid"

        private const val COST = 1
        private const val BLOCK_AMT = 8
        private const val HEAL_AMT = 2
        private const val BLOCK_UPG_AMT = 2
        private const val HEAL_UPG = 1
    }

    init {
        this.baseBlock = BLOCK_AMT
        this.block = this.baseBlock
        this.baseMagicNumber = HEAL_AMT
        this.magicNumber = this.baseMagicNumber
    }

    override fun upgrade() {
        if (!this.upgraded) {
            this.upgradeName()
            this.upgradeBlock(BLOCK_UPG_AMT)
            this.upgradeMagicNumber(HEAL_UPG)
        }
    }

    override fun makeCopy(): AbstractCard {
        return Bandaid()
    }

    override fun use(abstractPlayer: AbstractPlayer?, abstractMonster: AbstractMonster?) {
        AbstractDungeon.actionManager.addToBottom(GainBlockAction(abstractPlayer, abstractPlayer, this.block))
        AbstractDungeon.actionManager.addToBottom(HealAction(abstractPlayer, abstractPlayer, this.magicNumber))

    }


}
